src/wasmtime/crates/lind-multi-process/src/lib.rs

```diff
@@ -231,97 +212,50 @@ impl<T: Clone + Send + 'static + std::marker::Sync, U: Clone + Send + 'static +
         // get the stack pointer global
         let stack_pointer = caller.get_stack_pointer().unwrap();
 
+        // get the wasm stack top address
+        let stack_low_usr = caller.as_context().get_stack_top();
+
+        // we store the unwind at the top of the user stack
+        let unwind_data_start_usr = stack_low_usr;
+        let unwind_data_start_sys = address as u64 + unwind_data_start_usr;
+
         // start unwind
-        if let Some(asyncify_start_unwind_extern) = caller.get_export(ASYNCIFY_START_UNWIND) {
-            match asyncify_start_unwind_extern {
-                Extern::Func(asyncify_start_unwind) => {
-                    match asyncify_start_unwind.typed::<i32, ()>(&caller) {
-                        Ok(func) => {
-                            let unwind_pointer: u64 = 0;
-                            // 8 because we need to store unwind_data_start and unwind_data_end
-                            // at the beginning of the unwind stack as the parameter for asyncify_start_unwind
-                            // each of them are u64, so together is 8 bytes
-                            let unwind_data_start: u64 = unwind_pointer + 8;
-                            let unwind_data_end: u64 = stack_pointer as u64;
-    
-                            // store the parameter at the top of the stack
-                            unsafe {
-                                *(address as *mut u64) = unwind_data_start;
-                                *(address as *mut u64).add(1) = unwind_data_end;
-                            }
-                            
-                            // mark the start of unwind
-                            let _res = func.call(&mut caller, unwind_pointer as i32);
-                        }
-                        Err(err) => {
-                            println!("the signature of asyncify_start_unwind function is not correct: {:?}", err);
-                            return Ok(-1);
-                        }
-                    }
-                },
-                _ => {
-                    println!("asyncify_start_unwind export is not a function");
-                    return Ok(-1);
-                }
-            }
-        }
-        else {
-            println!("asyncify_start_unwind export not found");
-            return Ok(-1);
+        let asyncify_start_unwind_func = caller.get_asyncify_start_unwind().unwrap();
+
+        // store the parameter at the top of the stack
+        // we need to tell two parameters to Asyncify:
+        // unwind_data_start: the start address to store the unwind data.
+        // unwind_data_end: the end address of the avaliable space Asyncify could work with
+        // These two parameters are usually stored on the top of the unwind data.
+        // Below is a graph describing the entire user's stack layout
+        // -------------------------- <----- _usr (stack low)
```

Comment:
i like this diagram a lot as we discussed, its still a bit confusing that unwind_data_start and end are pointers stored at that address, perhaps we can make that a bit more clear by not depicting them in large regions